src/test/java/api/RelatedInstanceTest.java

```diff
@@ -0,0 +1,308 @@
+
+    var thirdNodJson = nod(anotherRelatedInstanceId)
+        .put(Instance.HRID_KEY, "in00000001")
+        .put(Instance.TITLE_KEY, "Another related instance")
+        .put(Instance.RELATED_INSTANCES_KEY, new JsonArray()
+            .add(createRelatedInstance(UUID.randomUUID(), anotherRelatedInstanceId, relatedInstanceId,
+                relatedInstanceTypeId)));
+
+    instancesClient.create(thirdNodJson);
+
+    var firstInstanceJson = instancesClient.getById(instanceId).getJson();
+    var firstInstanceRelatedInstancesJson = firstInstanceJson.getJsonArray(Instance.RELATED_INSTANCES_KEY);
+    var firstInstanceFirstRelatedInstanceJson = firstInstanceRelatedInstancesJson.getJsonObject(0);
+
+    assertThat(firstInstanceRelatedInstancesJson, notNullValue());
+    assertEquals(1, firstInstanceRelatedInstancesJson.size());
+
+    assertThat(firstInstanceRelatedInstancesJson, notNullValue());
+    assertThat(firstInstanceFirstRelatedInstanceJson.getString(RelatedInstance.INSTANCE_ID_KEY),
+        is(instanceId.toString()));
+    assertThat(firstInstanceFirstRelatedInstanceJson.getString(RelatedInstance.RELATED_INSTANCE_ID_KEY),
+        is(relatedInstanceId.toString()));
+
+    var secondInstanceJson = instancesClient.getById(relatedInstanceId).getJson();
+    var secondInstanceRelatedInstancesJson = secondInstanceJson.getJsonArray(Instance.RELATED_INSTANCES_KEY);
+    var secondInstanceFirstRelatedInstanceJson = secondInstanceRelatedInstancesJson.getJsonObject(0);
+    var secondInstanceSecondRelatedInstanceJson = secondInstanceRelatedInstancesJson.getJsonObject(1);
+
+    assertThat(secondInstanceRelatedInstancesJson, notNullValue());
+    assertEquals(2, secondInstanceRelatedInstancesJson.size());
+
+    assertThat(secondInstanceFirstRelatedInstanceJson, notNullValue());
+    assertThat(secondInstanceFirstRelatedInstanceJson.getString(RelatedInstance.INSTANCE_ID_KEY),
+        is(relatedInstanceId.toString()));
+
+    assertThat(secondInstanceSecondRelatedInstanceJson, notNullValue());
+    assertThat(secondInstanceSecondRelatedInstanceJson.getString(RelatedInstance.INSTANCE_ID_KEY),
+        is(relatedInstanceId.toString()));
+
+    if (secondInstanceFirstRelatedInstanceJson.getString(RelatedInstance.RELATED_INSTANCE_ID_KEY).equals(instanceId.toString())) {
+        assertThat(secondInstanceFirstRelatedInstanceJson.getString(RelatedInstance.RELATED_INSTANCE_ID_KEY),
+            is(instanceId.toString()));
+
+        assertThat(secondInstanceSecondRelatedInstanceJson.getString(RelatedInstance.RELATED_INSTANCE_ID_KEY),
+            is(anotherRelatedInstanceId.toString()));
+    } else {
+        assertThat(secondInstanceFirstRelatedInstanceJson.getString(RelatedInstance.RELATED_INSTANCE_ID_KEY),
+            is(anotherRelatedInstanceId.toString()));
+
+        assertThat(secondInstanceSecondRelatedInstanceJson.getString(RelatedInstance.RELATED_INSTANCE_ID_KEY),
+            is(instanceId.toString()));
+    }
```

Comment:
Why is this branching needed?

Is it to match up multiple related instance to the correct expectations?

Reply:
Because random UUID will be in a random order in the list.

Reply:
I have tried ordering the collection before setting the list on the instance, but it appears serialization is causing reorder.